stepper.c

```diff
@@ -30,13 +30,84 @@
 #include "report.h"
 #include "magazine.h"
 
+#include "spi.h"
+
 // Some useful constants.
 #define DT_SEGMENT (1.0/(ACCELERATION_TICKS_PER_SECOND*60.0)) // min/segment
 #define REQ_MM_INCREMENT_SCALAR 1.25
 #define RAMP_ACCEL 0
 #define RAMP_CRUISE 1
 #define RAMP_DECEL 2
 
+//SPI drivers
+#define SPI_ADDRESS_MASK        0x7
+#define SPI_RW_BIT              7
+// For use with the SPI driver chip
+typedef enum {
+  XTABLE,
+  YTABLE,
+  GRIPPER,
+  CAROUSEL
+} steppers_t;
+
+static const uint8_t scs_pin_lookup[4] = {
+  SCS_XTABLE_PIN,
+  SCS_YTABLE_PIN,
+  SCS_GRIPPER_PIN,
+  SCS_CAROUSEL_PIN
+  
+};
+
+// Initialization values for spi stepper drivers
+// Format: MSB, LSB
+static const uint8_t stepper_init_registers[4][18] = {
+  {
+    //XTABLE
+    0x0C, 0x11, // CTRL,   DTIME=11, ISGAIN=00, EXSTALL=0, MODE=0010, RSTEP=0, RDIR=0, ENBL=1
+    0x11, 0xFF, // TORQUE, SMPLTH=001, TORQUE=0xFF
+    0x20, 0x30, // OFF,    PWMMODE=0, TOFF=0x30
+    0x30, 0x80, // BLANK,  ABT=0, TBLANK=0x80
+    0x41, 0x10, // DECAY,  DECMOD=001, TDECAY=0x10
+    0x50, 0x40, // STALL,  VDIV=00, SDCNT=00, SDTHR=0x40
+    0x6A, 0x59, // DRIVE,  IDRIVEP=10, IDRIVEN=10, TDRIVEP=01, TDRIVEN=01, OCPDEG=10, OCPTH=01
+    0x70, 0x00  // STATUS, init all status flags to 0 
+  },
+  {
+    //YTABLE
+    0x0C, 0x11, // CTRL,   DTIME=11, ISGAIN=00, EXSTALL=0, MODE=0010, RSTEP=0, RDIR=0, ENBL=1	
+    0x11, 0xFF, // TORQUE, SMPLTH=001, TORQUE=0xFF
+    0x20, 0x30, // OFF,    PWMMODE=0, TOFF=0x30
+    0x30, 0x80, // BLANK,  ABT=0, TBLANK=0x80
+    0x41, 0x10, // DECAY,  DECMOD=001, TDECAY=0x10
+    0x50, 0x40, // STALL,  VDIV=00, SDCNT=00, SDTHR=0x40
+    0x6A, 0x59, // DRIVE,  IDRIVEP=10, IDRIVEN=10, TDRIVEP=01, TDRIVEN=01, OCPDEG=10, OCPTH=01
+    0x70, 0x00  // STATUS, init all status flags to 0 
+  },
+  {
+    //GRIPPER
+    0x0C, 0x11, //  CTRL,   DTIME=11, ISGAIN=00, EXSTALL=0, MODE=0010, RSTEP=0, RDIR=0, ENBL=1  
+    0x11, 0xFF, //  TORQUE, SMPLTH=001, TORQUE=0xFF
+    0x20, 0x30, //  OFF,    PWMMODE=0, TOFF=0x30
+    0x30, 0x80, //  BLANK,  ABT=0, TBLANK=0x80
+    0x41, 0x10, //  DECAY,  DECMOD=001, TDECAY=0x10
+    0x50, 0x40, //  STALL,  VDIV=00, SDCNT=00, SDTHR=0x40
+    0x6A, 0x59, //  DRIVE,  IDRIVEP=10, IDRIVEN=10, TDRIVEP=01, TDRIVEN=01, OCPDEG=10, OCPTH=01   
+    0x70, 0x00  //  STATUS, init all status flags to 0 
+
+  },
+  {
+    //CAROUSEL
+    0x0C, 0x11, //  CTRL,   DTIME=11, ISGAIN=00, EXSTALL=0, MODE=0010, RSTEP=0, RDIR=0, ENBL=1  
+    0x11, 0xFF, //  TORQUE, SMPLTH=001, TORQUE=0xFF
+    0x20, 0x30, //  OFF,    PWMMODE=0, TOFF=0x30
+    0x30, 0x80, //  BLANK,  ABT=0, TBLANK=0x80
+    0x41, 0x10, //  DECAY,  DECMOD=001, TDECAY=0x10
+    0x50, 0x40, //  STALL,  VDIV=00, SDCNT=00, SDTHR=0x40
+    0x6A, 0x59, //  DRIVE,  IDRIVEP=10, IDRIVEN=10, TDRIVEP=01, TDRIVEN=01, OCPDEG=10, OCPTH=01   
+    0x70, 0x00  //  STATUS, init all status flags to 0 
+  }
+};
```

Comment:
(Not necessarily now), this should be a single base list with overrides to conserve space.